packages/arb-token-bridge-ui/src/hooks/useArbTokenBridge.ts

```diff
@@ -157,81 +247,13 @@ export const useArbTokenBridge = (
     const l1ChainID = l1.network.id
     const l2ChainID = l2.network.id
 
-    const bridgeTokensToAdd: ContractStorage<ERC20BridgeToken> = {}
-
-    const candidateUnbridgedTokensToAdd: ERC20BridgeToken[] = []
-
-    for (const tokenData of arbTokenList.tokens) {
-      const { address, name, symbol, extensions, decimals, logoURI, chainId } =
-        tokenData
-
-      if (![l1ChainID, l2ChainID].includes(chainId)) {
-        continue
-      }
-
-      const bridgeInfo = (() => {
-        // TODO: parsing the token list format could be from arbts or the tokenlist package
-        interface Extensions {
-          bridgeInfo: {
-            [chainId: string]: {
-              tokenAddress: string
-              originBridgeAddress: string
-              destBridgeAddress: string
-            }
-          }
-        }
-        const isExtensions = (obj: any): obj is Extensions => {
-          if (!obj) return false
-          if (!obj['bridgeInfo']) return false
-          return Object.keys(obj['bridgeInfo'])
-            .map(key => obj['bridgeInfo'][key])
-            .every(
-              e =>
-                e &&
-                'tokenAddress' in e &&
-                'originBridgeAddress' in e &&
-                'destBridgeAddress' in e
-            )
-        }
-        if (!isExtensions(extensions)) {
-          return null
-        } else {
-          return extensions.bridgeInfo
-        }
-      })()
-
-      if (bridgeInfo) {
-        const l1Address = bridgeInfo[l1NetworkID]?.tokenAddress.toLowerCase()
-
-        if (!l1Address) {
-          return
-        }
-
-        bridgeTokensToAdd[l1Address] = {
-          name,
-          type: TokenType.ERC20,
-          symbol,
-          address: l1Address,
-          l2Address: address.toLowerCase(),
-          decimals,
-          logoURI,
-          listIds: new Set([listId])
-        }
-      }
-      // save potentially unbridged L1 tokens:
-      // stopgap: giant lists (i.e., CMC list) currently severaly hurts page performace, so for now we only add the bridged tokens
-      else if (arbTokenList.tokens.length < 1000) {
-        candidateUnbridgedTokensToAdd.push({
-          name,
-          type: TokenType.ERC20,
-          symbol,
-          address: address.toLowerCase(),
-          decimals,
-          logoURI,
-          listIds: new Set([listId])
-        })
-      }
-    }
+    const { bridgeTokensToAdd, candidateUnbridgedTokensToAdd } =
+      await processTokensFromList({
+        arbTokenList,
+        listId,
+        parentChainId: l1ChainID,
+        childChainId: l2ChainID
+      })
```

Comment:
had to move the processing code out, otherwise we will have to mock all the external and internal hooks, e.g. useAccount from wagmi, useQueryParams etc in order to render `useArbTokenBridge` in the test file, which points to an issue of this hook and make it pretty much untestable